### PR TITLE
[debezium] Bump debezium version to 1.9.7.Final

### PR DESCRIPTION
**Pull request by tigrulya-exe**
This PR upgrades Debezium version to 1.9.7.Final and uses @GOODBOY008 commit from #1236 as basis.

### CODE REVIEW

File: flink-connector-mongodb-cdc/src/main/java/com/ververica/cdc/connectors/mongodb/source/config/MongoDBSourceOptions.java [hunk 1]
```diff
@@ -90,6 +90,13 @@ public class MongoDBSourceOptions {
                     .defaultValue(1024)
                     .withDescription("The cursor batch size. Defaults to 1024.");
 
+    public static final ConfigOption<Integer> SNAPSHOT_BATCH_SIZE =
```
Comment (GOODBOY008) [inline flink-connector-mongodb-cdc/src/main/java/com/ververica/cdc/connectors/mongodb/source/config/MongoDBSourceOptions.java:93]: This feature should split into another pr .Btw, document should be updated too. cc @Jiabao-Sun

File: flink-connector-oracle-cdc/src/test/java/com/ververica/cdc/connectors/oracle/source/OracleSourceITCase.java [hunk 2]
```diff
@@ -146,7 +146,8 @@ private void testOracleParallelSource(
                                 + " 'table-name' = '%s',"
                                 + " 'scan.incremental.snapshot.enabled' = 'false',"
                                 + " 'debezium.log.mining.strategy' = 'online_catalog',"
-                                + " 'debezium.log.mining.continuous.mine' = 'true'"
+                                + " 'debezium.log.mining.continuous.mine' = 'true',"
+                                + " 'debezium.database.history.store.only.captured.tables.ddl' = 'true'"
```
Comment (GOODBOY008) [inline flink-connector-oracle-cdc/src/test/java/com/ververica/cdc/connectors/oracle/source/OracleSourceITCase.java:150]: We can remove config `[database.history.store.only.captured.tables.ddl](https://debezium.io/documentation/reference/1.9/connectors/oracle.html#oracle-property-database-history-store-only-captured-tables-ddl)` in test case.

**Review by GOODBOY008 — COMMENTED** [review]
@tigrulya-exe Thanks for your great contributions, fix e2e failure, I just left some comments.

File: flink-connector-debezium/src/main/java/com/ververica/cdc/debezium/DebeziumSourceFunction.java [hunk 3]
```diff
@@ -390,6 +390,9 @@ public void run(SourceContext<T> sourceContext) throws Exception {
         // and https://debezium.io/blog/2018/03/16/note-on-database-history-topic-configuration/
         properties.setProperty("database.history", determineDatabase().getCanonicalName());
 
+        // we need this in order not to lose any transaction during snapshot to streaming switch
+        properties.setProperty("internal.log.mining.transaction.snapshot.boundary.mode", "all");
```
Comment (GOODBOY008) [inline flink-connector-debezium/src/main/java/com/ververica/cdc/debezium/DebeziumSourceFunction.java:394]: This config option is just worked in `oracle-cdc`.But in flink cdc this config is useless when snapshot by flink cdc without using debezium snapshot.

File: flink-connector-mongodb-cdc/src/main/java/com/ververica/cdc/connectors/mongodb/source/config/MongoDBSourceOptions.java (hunk 1 above)
Comment (tigrulya-exe) [reply flink-connector-mongodb-cdc/src/main/java/com/ververica/cdc/connectors/mongodb/source/config/MongoDBSourceOptions.java]: Temporarily replaced with batch size. I'll create pr, when the current one will be merged.

File: flink-connector-oracle-cdc/src/test/java/com/ververica/cdc/connectors/oracle/source/OracleSourceITCase.java (hunk 2 above)
Comment (tigrulya-exe) [reply flink-connector-oracle-cdc/src/test/java/com/ververica/cdc/connectors/oracle/source/OracleSourceITCase.java:150]: But in this case, tests with `parallelismSnapshot: false` will take much longer to run because of capturing structure of system APEX tables. I think, that we don't need to monitor them during tests.

File: flink-connector-debezium/src/main/java/com/ververica/cdc/debezium/DebeziumSourceFunction.java (hunk 3 above)
Comment (tigrulya-exe) [reply flink-connector-debezium/src/main/java/com/ververica/cdc/debezium/DebeziumSourceFunction.java]: Oh, my bad, I misplaced it. Transferred it to `OracleSource#build()`. But, unfortunately, I don't understand second part of your comment. Catching pending transactions during snapshot finish works only in debezium and if incremental snapshot mode is enabled, then this functionality won't work, right. But implementing such behavior in incremental sources is beyond the scope of this pr.

Comment (GOODBOY008) [reply flink-connector-debezium/src/main/java/com/ververica/cdc/debezium/DebeziumSourceFunction.java]: I means `internal.log.mining.transaction.snapshot.boundary.mode` this config will not worked in parallelismSnapshot. And I think this config option can also keeped.

File: flink-connector-mongodb-cdc/src/main/java/com/ververica/cdc/connectors/mongodb/source/config/MongoDBSourceOptions.java (hunk 1 above)
Comment (GOODBOY008) [reply flink-connector-mongodb-cdc/src/main/java/com/ververica/cdc/connectors/mongodb/source/config/MongoDBSourceOptions.java]: > Temporarily replaced with batch size. I'll create pr, when the current one will be merged.

@tigrulya-exe MongoDB snapshot phase use cusor fetch , `batchSize` seems same to cusor fetch.

File: flink-connector-oracle-cdc/src/test/java/com/ververica/cdc/connectors/oracle/source/OracleSourceITCase.java (hunk 2 above)
Comment (GOODBOY008) [reply flink-connector-oracle-cdc/src/test/java/com/ververica/cdc/connectors/oracle/source/OracleSourceITCase.java:150]: Ok, It can speed up unit test. I got it.

File: flink-connector-debezium/src/main/java/com/ververica/cdc/debezium/DebeziumSourceFunction.java (hunk 3 above)
Comment (tigrulya-exe) [reply flink-connector-debezium/src/main/java/com/ververica/cdc/debezium/DebeziumSourceFunction.java]: Yes, now this config is enabled by default in non incremental oracle source

File: flink-connector-mongodb-cdc/src/main/java/com/ververica/cdc/connectors/mongodb/source/config/MongoDBSourceOptions.java (hunk 1 above)
Comment (tigrulya-exe) [reply flink-connector-mongodb-cdc/src/main/java/com/ververica/cdc/connectors/mongodb/source/config/MongoDBSourceOptions.java]: Do you mean not to introduce a new configuration parameter for the batch size at the snapshot stage and keep it as it is now (use batchSize of streaming for shapshot queue size)? Ok

**Comment by GOODBOY008** [comment]
@tigrulya-exe hi, Can you rebase to master branch and migrate sqlserver to dbz 1.9.7

**Comment by tigrulya-exe** [comment]
@GOODBOY008 Hi, I rebased to master and migrated new SqlServer incremental source to the debezium 1.9.7, please take a look. Btw, I have no idea why Github decided to automatically close PR, I reopened it.

**Comment by ruanhang1993** [comment]
Hi, @tigrulya-exe . Thanks a lot for the contribution.

#1823 will use Incremental Snapshot Framework for Postgres CDC Connector. We will try to merge it as soon as possible.
Could you help to migrate it to dbz 1.9.7 again after merging that PR?

**Comment by tigrulya-exe** [comment]
@ruanhang1993 Hello! Sure, I'll try to help.

**Comment by ruanhang1993** [comment]
> @ruanhang1993 Hello! Sure, I'll try to help.

Hi, @tigrulya-exe . Thanks a lot for your help.
PG CDC PR has already been merged. There are some failed test for pg cdc. We will resolve them in another PR.
Now you could continue developing this PR. Thanks~

ps: Oracle cdc bug fix #2215 also introduces a debezium class.

**Comment by GOODBOY008** [comment]
@tigrulya-exe Thanks for your great contributions. @ruanhang1993 I think pg test fail should not be fixed in this pr.

**Review by GOODBOY008 — APPROVED** [review]

**Review by ruanhang1993 — APPROVED** [review]
@tigrulya-exe Thanks a lot for your great work. We will resolve PG cdc tests in another PR.
